CharlieBackend.Api/Middlewares/IsAccountActiveMiddleware.cs

```diff
@@ -36,55 +38,25 @@ public async Task InvokeAsync(HttpContext context, IAccountService accountServic
             }
             else
             {
-                string authHeader = context.Request.Headers["Authorization"];
+                var currentEmail = currentUserService.Email;
 
-                if (authHeader != null)
+                var isActive = await accountService.IsAccountActiveAsync(currentEmail);
+                
+                if (isActive == null)
                 {
-                    var handler = new JwtSecurityTokenHandler();
-                    authHeader = authHeader.ToString().Replace("Bearer ", "");
+                    context.Response.StatusCode = 401;
 
-                    var jsonToken = handler.ReadToken(authHeader);
-                    var tokenS = handler.ReadToken(authHeader) as JwtSecurityToken;
-
-                    var email = tokenS.Claims.First(claim => claim.Type == "Email").Value;
-
-                    var isActive = await accountService.IsAccountActiveAsync(email);
-
-                    if (isActive == null)
-                    {
-                        context.Response.StatusCode = 401;
-
-                        await context.Response.WriteAsync("Need to sign in.");
-                    }
-
-                    if ((bool)!isActive)
-                    {
-                        context.Response.StatusCode = 401;
-
-                        await context.Response.WriteAsync("Account is not active!");
-                    }
-
-                    if (context.Request.Path.Value.Contains("lessons"))
-                    {
-                        var role = tokenS.Claims
-                                .First(claim => claim.Type == ClaimsIdentity.DefaultRoleClaimType).Value;
-
-                        if (role == "2")
-                        {
-                            var id = tokenS.Claims.First(claim => claim.Type == "Id").Value;
-
-                            context.Items["mentorId"] = id;
-                        }
-                    }
-
-                    await _next.Invoke(context);
+                    await context.Response.WriteAsync("Need to sign in.");
                 }
-                else
+                    
+                if ((bool)!isActive)
                 {
-                    context.Response.StatusCode = 400;
+                     context.Response.StatusCode = StatusCodes.Status403Forbidden;
 
-                    await context.Response.WriteAsync("Bad token.");
-                }
+                     await context.Response.WriteAsync("Account is deactivated");
```

Comment:
Since you are refactoring this middleware, moving these error messages to string constants would probably be a good practice too.